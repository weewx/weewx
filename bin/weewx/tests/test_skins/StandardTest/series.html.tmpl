#errorCatcher Echo
## Copyright 2009-2021 Tom Keffer
## Distributed under terms of GPLv3.  See LICENSE.txt for your rights.
##
## This template tests tags used for generating series
##
##
## Specifying an encoding of UTF-8 is usually safe:
#encoding UTF-8
##
<!DOCTYPE HTML>
<html lang="en">
  <head>
      <meta charset="UTF-8">
      <title>Test the ".series" tag</title>
      <style>
          p {
              font-weight: bold;
          }
          table, th, td {
              border: 1px solid black;
              border-collapse: collapse;
          }    </style>
  </head>

  <body>
    <h2>Unaggregated series </h2>
	<p>Unaggregated series in json: \$day.outTemp.series.round(5).json():</p>
    <pre>$day.outTemp.series.round(5).json()</pre>

	<p>Unaggregated series, in json with just start times: \$day.outTemp.series(time_series='start').round(5).json:</p>
    <pre>$day.outTemp.series(time_series='start').round(5).json</pre>

	<p>Unaggregated series, in json with start times in milliseconds: \$day.outTemp.series(time_series='start', time_unit='unix_epoch_ms').round(5).json:</p>
    <pre>$day.outTemp.series(time_series='start', time_unit='unix_epoch_ms').round(5).json</pre>

    <p>Unaggregated series, in json, in degrees C, rounded to 5 decimal places: \$day.outTemp.series.degree_C.round(5).json</p>
    <pre>$day.outTemp.series.degree_C.round(5).json</pre>

    <p>Unaggregated series, as a formatted string (<em>not JSON</em>) : \$day.outTemp.series:</p>
    <pre>$day.outTemp.series</pre>

    <p>Unaggregated series, start time only, as a formatted string (<em>not JSON</em>) : \$day.outTemp.series(time_series='start'):</p>
    <pre>$day.outTemp.series(time_series='start')</pre>

    <p>Unaggregated series, stop time only, as a formatted string (<em>not JSON</em>) : \$day.outTemp.series(time_series='stop'):</p>
    <pre>$day.outTemp.series(time_series='stop')</pre>

    <p>Unaggregated series, by column, as a formatted string (<em>not JSON</em>) \$day.outTemp.series.format(order_by='column'):</p>
    <pre>$day.outTemp.series.format(order_by='column')</pre>

    <p>Unaggregated series, by column, start times only, as a formatted string (<em>not JSON</em>) \$day.outTemp.series(time_series='start').format(order_by='column'):</p>
    <pre>$day.outTemp.series(time_series='start').format(order_by='column')</pre>

    <p>Unaggregated series, by column, stop times only, as a formatted string (<em>not JSON</em>) \$day.outTemp.series(time_series='stop').format(order_by='column'):</p>
    <pre>$day.outTemp.series(time_series='stop').format(order_by='column')</pre>

	<hr/>

	<h2>Aggregated series</h2>
	<p>Aggregated series: \$month.outTemp.series(aggregate_type='max', aggregate_interval=86400).round(5).json():</p>
    <pre>$month.outTemp.series(aggregate_type='max', aggregate_interval=86400).round(5)json()</pre>

	<p>Using shortcut 'day': \$month.outTemp.series(aggregate_type='max', aggregate_interval='day').round(5).json():</p>
    <pre>$month.outTemp.series(aggregate_type='max', aggregate_interval='day').round(5).json()</pre>

	<p>Order by column: \$month.outTemp.series(aggregate_type='max', aggregate_interval=86400).round(5).json(order_by="column"):</p>
    <pre>$month.outTemp.series(aggregate_type='max', aggregate_interval=86400).round(5).json(order_by="column")</pre>

    <p>Aggregated series, using \$jsonize():
    \$jsonize(\$zip(\$min.start.unix_epoch_ms.raw, \$min.data.degree_C.round(2).raw, \$max.data.degree_C.round(2).raw))
    #set $min = $month.outTemp.series(aggregate_type='min', aggregate_interval='day')
    #set $max = $month.outTemp.series(aggregate_type='max', aggregate_interval='day')
    <pre>
    $jsonize($zip($min.start.unix_epoch_ms.raw, $min.data.degree_C.round(2).raw, $max.data.degree_C.round(2).raw))
    </pre>


    <hr/>

    <h2>Aggregated series of wind vectors</h2>

    <p>Aggregated wind seriex (<em>not JSON)</em>), complex notation
        \$month.windvec.series(aggregate_type='max', aggregate_interval='day')</p>
    <pre>$month.windvec.series(aggregate_type='max', aggregate_interval='day')</pre>

    <p>Aggregated wind seriex (<em>not JSON)</em>), complex notation with formatting
        \$month.windvec.series(aggregate_type='max', aggregate_interval='day').format("%.5f")</p>
    <pre>$month.windvec.series(aggregate_type='max', aggregate_interval='day').format("%.5f")</pre>

    <p>Aggregated wind seriex (<em>not JSON</em>), polar notation
        \$month.windvec.series(aggregate_type='max', aggregate_interval='day').polar.format("%.5f")</p>
    <pre>$month.windvec.series(aggregate_type='max', aggregate_interval='day').polar.format("%.5f")</pre>

    <hr/>

    <h2>Aggregated series of wind vectors with conversions</h2>

    <p>Starting series: \$month.windvec.series(aggregate_type='max', aggregate_interval='day').json()</p>
    <pre>
    $month.windvec.series(aggregate_type='max', aggregate_interval='day').json()
    </pre>

    <p> X-component: \$month.windvec.series(aggregate_type='max', aggregate_interval='day').x.json()</p>
    <pre>
    $month.windvec.series(aggregate_type='max', aggregate_interval='day').x.json()
    </pre>

    <p> x-component, in knots: \$month.windvec.series(aggregate_type='max', aggregate_interval='day').x.knot.json()</p>
    <pre>
    $month.windvec.series(aggregate_type='max', aggregate_interval='day').x.knot.json()
    </pre>

    <p>Y-component: \$month.windvec.series(aggregate_type='max', aggregate_interval='day').y.json()</p>
    <pre>
    $month.windvec.series(aggregate_type='max', aggregate_interval='day').y.json()
    </pre>

    <p>magnitude: \$month.windvec.series(aggregate_type='max', aggregate_interval='day').magnitude.json()</p>
    <pre>
    $month.windvec.series(aggregate_type='max', aggregate_interval='day').magnitude.json()
    </pre>

    <p>direction: \$month.windvec.series(aggregate_type='max', aggregate_interval='day').direction.json()</p>
    <pre>
    $month.windvec.series(aggregate_type='max', aggregate_interval='day').direction.json()
    </pre>

    <p>polar: \$month.windvec.series(aggregate_type='max', aggregate_interval='day').polar.json()</p>
    <pre>
    $month.windvec.series(aggregate_type='max', aggregate_interval='day').polar.json()
    </pre>

    <p>polar in knots: \$month.windvec.series(aggregate_type='max', aggregate_interval='day').knot.polar.round(5).json()</p>
    <pre>
    $month.windvec.series(aggregate_type='max', aggregate_interval='day').knot.polar.round(5).json()
    </pre>

  <hr/>
  <h2>Iterate over an aggregated series</h2>
  <pre>
    \#for (\$start, \$stop, \$data) in \$month.outTemp.series(aggregate_type='max', aggregate_interval='day')
      ...
    \#end for
  </pre>

    <table>
      <tr>
        <td>Start date</td>
        <td>Stop date</td>
        <td>Max temperature</td>
      </tr>
    #for ($start, $stop, $data) in $month.outTemp.series(aggregate_type='max', aggregate_interval='day')
      <tr>
        <td>$start.format("%Y-%m-%d")</td>
        <td>$stop.format("%Y-%m-%d")</td>
        <td>$data.format("%.2f")</td>
      </tr>
    #end for
    </table>

  </body>

</html>
