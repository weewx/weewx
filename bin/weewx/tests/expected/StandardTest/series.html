<!DOCTYPE HTML>
<html lang="en">
  <head>
      <meta charset="UTF-8">
      <title>Test the ".series" tag</title>
      <style>
          p {
              font-weight: bold;
          }
          table, th, td {
              border: 1px solid black;
              border-collapse: collapse;
          }    </style>
  </head>

  <body>
    <h2>Unaggregated series </h2>
	<p>Unaggregated series in json: $day.outTemp.series.round(5).json():</p>
    <pre>[[1283497200, 1283499000, 46.83544], [1283499000, 1283500800, 44.85591], [1283500800, 1283502600, 43.11835], [1283502600, 1283504400, null], [1283504400, 1283506200, 40.48257], [1283506200, 1283508000, 39.62901], [1283508000, 1283509800, 39.106], [1283509800, 1283511600, 38.92225], [1283511600, 1283513400, 39.08072], [1283513400, 1283515200, 39.57846], [1283515200, 1283517000, 40.40674], [1283517000, 1283518800, 41.55117], [1283518800, 1283520600, 42.99197], [1283520600, 1283522400, 44.70425], [1283522400, 1283524200, 46.6585], [1283524200, 1283526000, 48.82107], [1283526000, 1283527800, 51.15474], [1283527800, 1283529600, 53.61937], [1283529600, 1283531400, 56.17256], [1283531400, 1283533200, 58.77042], [1283533200, 1283535000, 61.36827], [1283535000, 1283536800, 63.92146]]</pre>

	<p>Unaggregated series, in json with just start times: $day.outTemp.series(time_series='start').round(5).json:</p>
    <pre>[[1283497200, 46.83544], [1283499000, 44.85591], [1283500800, 43.11835], [1283502600, null], [1283504400, 40.48257], [1283506200, 39.62901], [1283508000, 39.106], [1283509800, 38.92225], [1283511600, 39.08072], [1283513400, 39.57846], [1283515200, 40.40674], [1283517000, 41.55117], [1283518800, 42.99197], [1283520600, 44.70425], [1283522400, 46.6585], [1283524200, 48.82107], [1283526000, 51.15474], [1283527800, 53.61937], [1283529600, 56.17256], [1283531400, 58.77042], [1283533200, 61.36827], [1283535000, 63.92146]]</pre>

	<p>Unaggregated series, in json with start times in milliseconds: $day.outTemp.series(time_series='start', time_unit='unix_epoch_ms').round(5).json:</p>
    <pre>[[1283497200000, 46.83544], [1283499000000, 44.85591], [1283500800000, 43.11835], [1283502600000, null], [1283504400000, 40.48257], [1283506200000, 39.62901], [1283508000000, 39.106], [1283509800000, 38.92225], [1283511600000, 39.08072], [1283513400000, 39.57846], [1283515200000, 40.40674], [1283517000000, 41.55117], [1283518800000, 42.99197], [1283520600000, 44.70425], [1283522400000, 46.6585], [1283524200000, 48.82107], [1283526000000, 51.15474], [1283527800000, 53.61937], [1283529600000, 56.17256], [1283531400000, 58.77042], [1283533200000, 61.36827], [1283535000000, 63.92146]]</pre>

    <p>Unaggregated series, in json, in degrees C, rounded to 5 decimal places: $day.outTemp.series.degree_C.round(5).json</p>
    <pre>[[1283497200, 1283499000, 8.24191], [1283499000, 1283500800, 7.14217], [1283500800, 1283502600, 6.17686], [1283502600, 1283504400, null], [1283504400, 1283506200, 4.71254], [1283506200, 1283508000, 4.23834], [1283508000, 1283509800, 3.94778], [1283509800, 1283511600, 3.8457], [1283511600, 1283513400, 3.93373], [1283513400, 1283515200, 4.21025], [1283515200, 1283517000, 4.67041], [1283517000, 1283518800, 5.30621], [1283518800, 1283520600, 6.10665], [1283520600, 1283522400, 7.05791], [1283522400, 1283524200, 8.14361], [1283524200, 1283526000, 9.34504], [1283526000, 1283527800, 10.64152], [1283527800, 1283529600, 12.01076], [1283529600, 1283531400, 13.4292], [1283531400, 1283533200, 14.87245], [1283533200, 1283535000, 16.31571], [1283535000, 1283536800, 17.73414]]</pre>

    <p>Unaggregated series, as a formatted string (<em>not JSON</em>) : $day.outTemp.series:</p>
    <pre>00:00, 00:30, 46.8&#176;F
00:30, 01:00, 44.9&#176;F
01:00, 01:30, 43.1&#176;F
01:30, 02:00,    N/A
02:00, 02:30, 40.5&#176;F
02:30, 03:00, 39.6&#176;F
03:00, 03:30, 39.1&#176;F
03:30, 04:00, 38.9&#176;F
04:00, 04:30, 39.1&#176;F
04:30, 05:00, 39.6&#176;F
05:00, 05:30, 40.4&#176;F
05:30, 06:00, 41.6&#176;F
06:00, 06:30, 43.0&#176;F
06:30, 07:00, 44.7&#176;F
07:00, 07:30, 46.7&#176;F
07:30, 08:00, 48.8&#176;F
08:00, 08:30, 51.2&#176;F
08:30, 09:00, 53.6&#176;F
09:00, 09:30, 56.2&#176;F
09:30, 10:00, 58.8&#176;F
10:00, 10:30, 61.4&#176;F
10:30, 11:00, 63.9&#176;F</pre>

    <p>Unaggregated series, start time only, as a formatted string (<em>not JSON</em>) : $day.outTemp.series(time_series='start'):</p>
    <pre>00:00, 46.8&#176;F
00:30, 44.9&#176;F
01:00, 43.1&#176;F
01:30,    N/A
02:00, 40.5&#176;F
02:30, 39.6&#176;F
03:00, 39.1&#176;F
03:30, 38.9&#176;F
04:00, 39.1&#176;F
04:30, 39.6&#176;F
05:00, 40.4&#176;F
05:30, 41.6&#176;F
06:00, 43.0&#176;F
06:30, 44.7&#176;F
07:00, 46.7&#176;F
07:30, 48.8&#176;F
08:00, 51.2&#176;F
08:30, 53.6&#176;F
09:00, 56.2&#176;F
09:30, 58.8&#176;F
10:00, 61.4&#176;F
10:30, 63.9&#176;F</pre>

    <p>Unaggregated series, stop time only, as a formatted string (<em>not JSON</em>) : $day.outTemp.series(time_series='stop'):</p>
    <pre>00:30, 46.8&#176;F
01:00, 44.9&#176;F
01:30, 43.1&#176;F
02:00,    N/A
02:30, 40.5&#176;F
03:00, 39.6&#176;F
03:30, 39.1&#176;F
04:00, 38.9&#176;F
04:30, 39.1&#176;F
05:00, 39.6&#176;F
05:30, 40.4&#176;F
06:00, 41.6&#176;F
06:30, 43.0&#176;F
07:00, 44.7&#176;F
07:30, 46.7&#176;F
08:00, 48.8&#176;F
08:30, 51.2&#176;F
09:00, 53.6&#176;F
09:30, 56.2&#176;F
10:00, 58.8&#176;F
10:30, 61.4&#176;F
11:00, 63.9&#176;F</pre>

    <p>Unaggregated series, by column, as a formatted string (<em>not JSON</em>) $day.outTemp.series.format(order_by='column'):</p>
    <pre>00:00, 00:30, 01:00, 01:30, 02:00, 02:30, 03:00, 03:30, 04:00, 04:30, 05:00, 05:30, 06:00, 06:30, 07:00, 07:30, 08:00, 08:30, 09:00, 09:30, 10:00, 10:30
00:30, 01:00, 01:30, 02:00, 02:30, 03:00, 03:30, 04:00, 04:30, 05:00, 05:30, 06:00, 06:30, 07:00, 07:30, 08:00, 08:30, 09:00, 09:30, 10:00, 10:30, 11:00
46.8&#176;F, 44.9&#176;F, 43.1&#176;F,    N/A, 40.5&#176;F, 39.6&#176;F, 39.1&#176;F, 38.9&#176;F, 39.1&#176;F, 39.6&#176;F, 40.4&#176;F, 41.6&#176;F, 43.0&#176;F, 44.7&#176;F, 46.7&#176;F, 48.8&#176;F, 51.2&#176;F, 53.6&#176;F, 56.2&#176;F, 58.8&#176;F, 61.4&#176;F, 63.9&#176;F</pre>

    <p>Unaggregated series, by column, start times only, as a formatted string (<em>not JSON</em>) $day.outTemp.series(time_series='start').format(order_by='column'):</p>
    <pre>00:00, 00:30, 01:00, 01:30, 02:00, 02:30, 03:00, 03:30, 04:00, 04:30, 05:00, 05:30, 06:00, 06:30, 07:00, 07:30, 08:00, 08:30, 09:00, 09:30, 10:00, 10:30
46.8&#176;F, 44.9&#176;F, 43.1&#176;F,    N/A, 40.5&#176;F, 39.6&#176;F, 39.1&#176;F, 38.9&#176;F, 39.1&#176;F, 39.6&#176;F, 40.4&#176;F, 41.6&#176;F, 43.0&#176;F, 44.7&#176;F, 46.7&#176;F, 48.8&#176;F, 51.2&#176;F, 53.6&#176;F, 56.2&#176;F, 58.8&#176;F, 61.4&#176;F, 63.9&#176;F</pre>

    <p>Unaggregated series, by column, stop times only, as a formatted string (<em>not JSON</em>) $day.outTemp.series(time_series='stop').format(order_by='column'):</p>
    <pre>00:30, 01:00, 01:30, 02:00, 02:30, 03:00, 03:30, 04:00, 04:30, 05:00, 05:30, 06:00, 06:30, 07:00, 07:30, 08:00, 08:30, 09:00, 09:30, 10:00, 10:30, 11:00
46.8&#176;F, 44.9&#176;F, 43.1&#176;F,    N/A, 40.5&#176;F, 39.6&#176;F, 39.1&#176;F, 38.9&#176;F, 39.1&#176;F, 39.6&#176;F, 40.4&#176;F, 41.6&#176;F, 43.0&#176;F, 44.7&#176;F, 46.7&#176;F, 48.8&#176;F, 51.2&#176;F, 53.6&#176;F, 56.2&#176;F, 58.8&#176;F, 61.4&#176;F, 63.9&#176;F</pre>

	<hr/>

	<h2>Aggregated series</h2>
	<p>Aggregated series: $month.outTemp.series(aggregate_type='max', aggregate_interval=86400).round(5).json():</p>
    <pre>[[1283324400, 1283410800, 79.82582], [1283410800, 1283497200, 79.22488], [1283497200, 1283583600, 63.92146]]</pre>

	<p>Using shortcut 'day': $month.outTemp.series(aggregate_type='max', aggregate_interval='day').round(5).json():</p>
    <pre>[[1283324400, 1283410800, 79.82582], [1283410800, 1283497200, 79.22488], [1283497200, 1283583600, 63.92146]]</pre>

	<p>Order by column: $month.outTemp.series(aggregate_type='max', aggregate_interval=86400).round(5).json(order_by="column"):</p>
    <pre>[[1283324400, 1283410800, 1283497200], [1283410800, 1283497200, 1283583600], [79.82582, 79.22488, 63.92146]]</pre>

    <p>Aggregated series, using $jsonize():
    $jsonize($zip($min.start.unix_epoch_ms.raw, $min.data.degree_C.round(2).raw, $max.data.degree_C.round(2).raw))
    <pre>
    [[1283324400000, 4.51, 26.57], [1283410800000, 4.18, 26.24], [1283497200000, 3.85, 17.73]]
    </pre>


    <hr/>

    <h2>Aggregated series of wind vectors</h2>

    <p>Aggregated wind seriex (<em>not JSON)</em>), complex notation
        $month.windvec.series(aggregate_type='max', aggregate_interval='day')</p>
    <pre>01-Sep-2010 00:00, 02-Sep-2010 00:00, (-1, 9) mph
02-Sep-2010 00:00, 03-Sep-2010 00:00, (20, 1) mph
03-Sep-2010 00:00, 04-Sep-2010 00:00, (20, 0) mph</pre>

    <p>Aggregated wind seriex (<em>not JSON)</em>), complex notation with formatting
        $month.windvec.series(aggregate_type='max', aggregate_interval='day').format("%.5f")</p>
    <pre>01-Sep-2010 00:00, 02-Sep-2010 00:00, (-0.61126, 9.32596) mph
02-Sep-2010 00:00, 03-Sep-2010 00:00, (19.93581, 1.30666) mph
03-Sep-2010 00:00, 04-Sep-2010 00:00, (20.00000, 0.00000) mph</pre>

    <p>Aggregated wind seriex (<em>not JSON</em>), polar notation
        $month.windvec.series(aggregate_type='max', aggregate_interval='day').polar.format("%.5f")</p>
    <pre>01-Sep-2010 00:00, 02-Sep-2010 00:00, (9.34597 mph, 356&#176;)
02-Sep-2010 00:00, 03-Sep-2010 00:00, (19.97859 mph, 86&#176;)
03-Sep-2010 00:00, 04-Sep-2010 00:00, (20.00000 mph, 90&#176;)</pre>

    <hr/>

    <h2>Aggregated series of wind vectors with conversions</h2>

    <p>Starting series: $month.windvec.series(aggregate_type='max', aggregate_interval='day').json()</p>
    <pre>
    [[1283324400, 1283410800, [-0.6112556182879079, 9.325958562961368]], [1283410800, 1283497200, [19.935813307366608, 1.3066622382029147]], [1283497200, 1283583600, [20.0, 0.0]]]
    </pre>

    <p> X-component: $month.windvec.series(aggregate_type='max', aggregate_interval='day').x.json()</p>
    <pre>
    [[1283324400, 1283410800, -0.6112556182879079], [1283410800, 1283497200, 19.935813307366608], [1283497200, 1283583600, 20.0]]
    </pre>

    <p> x-component, in knots: $month.windvec.series(aggregate_type='max', aggregate_interval='day').x.knot.json()</p>
    <pre>
    [[1283324400, 1283410800, -0.5311666100812127], [1283410800, 1283497200, 17.323748129049026], [1283497200, 1283583600, 17.379524840000002]]
    </pre>

    <p>Y-component: $month.windvec.series(aggregate_type='max', aggregate_interval='day').y.json()</p>
    <pre>
    [[1283324400, 1283410800, 9.325958562961368], [1283410800, 1283497200, 1.3066622382029147], [1283497200, 1283583600, 0.0]]
    </pre>

    <p>magnitude: $month.windvec.series(aggregate_type='max', aggregate_interval='day').magnitude.json()</p>
    <pre>
    [[1283324400, 1283410800, 9.345969], [1283410800, 1283497200, 19.978589], [1283497200, 1283583600, 20.0]]
    </pre>

    <p>direction: $month.windvec.series(aggregate_type='max', aggregate_interval='day').direction.json()</p>
    <pre>
    [[1283324400, 1283410800, 356.25], [1283410800, 1283497200, 86.25], [1283497200, 1283583600, 90.0]]
    </pre>

    <p>polar: $month.windvec.series(aggregate_type='max', aggregate_interval='day').polar.json()</p>
    <pre>
    [[1283324400, 1283410800, [9.345969, 356.25]], [1283410800, 1283497200, [19.978589, 86.25]], [1283497200, 1283583600, [20.0, 90.0]]]
    </pre>

    <p>polar in knots: $month.windvec.series(aggregate_type='max', aggregate_interval='day').knot.polar.round(5).json()</p>
    <pre>
    [[1283324400, 1283410800, [8.12143, 356.25]], [1283410800, 1283497200, [17.36092, 86.25]], [1283497200, 1283583600, [17.37952, 90.0]]]
    </pre>

  <hr/>
  <h2>Iterate over an aggregated series</h2>
  <pre>
    #for ($start, $stop, $data) in $month.outTemp.series(aggregate_type='max', aggregate_interval='day')
      ...
    #end for
  </pre>

    <table>
      <tr>
        <td>Start date</td>
        <td>Stop date</td>
        <td>Max temperature</td>
      </tr>
      <tr>
        <td>2010-09-01</td>
        <td>2010-09-02</td>
        <td>79.83&#176;F</td>
      </tr>
      <tr>
        <td>2010-09-02</td>
        <td>2010-09-03</td>
        <td>79.22&#176;F</td>
      </tr>
      <tr>
        <td>2010-09-03</td>
        <td>2010-09-04</td>
        <td>63.92&#176;F</td>
      </tr>
    </table>

  </body>

</html>
